### DISCUSSION TITLE
Does “proxy mode” support client SSL certificates?

### DISCUSSION THREAD

**Discussion by Infarinato** [mockoon/mockoon#730] [Help]
Does “proxy mode” support _client_ SSL certificates? It would appear not, but maybe I’m missing something… If not, could this be taken into consideration as a future enhancement?

Thanks.

**Comment by 255kb** [comment]
I am not sure what you mean by "supporting client cert", could you please elaborate on what you would like to achieve? Thanks :)

**Reply by Infarinato** [reply]
I believe it’s sometimes called “2-way TLS” and is an industry-standard way to secure API endpoints by attaching a certificate to each request, which must be trusted by the API: https://github.com/mock-server/mockserver/issues/357#issuecomment-344177730. 😊

**Reply by 255kb** [reply]
Ok I see, this is not supported and wasn't requested until now. You can open an issue to see if it gathers some interest from other users. I cannot guarantee that we will implement this in the future or give any ETA.

**Comment by krnbr** [comment]
have covered the basic idea of it as part of the PR here - https://github.com/mockoon/mockoon/pull/1604
